app/controllers/admin/administrative_events_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Admin::AdministrativeEventsController < ApplicationController
+  before_action :require_can_add_administrative_event!
+  before_action :load_event, only: [:edit, :update]
+  
+  def index
+    @events = administrative_event_source.order(date: :desc).page(params[:page]).per(1)
+  end
+
+  def create
+    @event = administrative_event_source.create(administrative_event_params.merge({
+      user_id: current_user.id,
+    }))
+    respond_with(@event, location: admin_administrative_events_path)
+  end
+  
+  def new
+    @event = administrative_event_source.new 
+  end
+  
+  def edit
+  end
+
+  def update
+    @event.update(administrative_event_params)
+    respond_with(@event, location: admin_administrative_events_path)
+  end
+  
+  def destroy
+    @event = administrative_event_source.find params[:id]
```

Comment:
add destroy to the before_action above and line 29 can come out.
Then the second line can mirror update respond_with